org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SemanticTokensHandler.java

```diff
@@ -11,37 +11,48 @@
  *     Microsoft Corporation - initial API and implementation
  *******************************************************************************/
 
-package org.eclipse.jdt.ls.core.internal.commands;
+package org.eclipse.jdt.ls.core.internal.handlers;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.jdt.core.ITypeRoot;
 import org.eclipse.jdt.core.dom.CompilationUnit;
 import org.eclipse.jdt.core.manipulation.CoreASTProvider;
 import org.eclipse.jdt.ls.core.internal.JDTUtils;
-import org.eclipse.jdt.ls.core.internal.JobHelpers;
-import org.eclipse.jdt.ls.core.internal.handlers.DocumentLifeCycleHandler;
-import org.eclipse.jdt.ls.core.internal.semantictokens.SemanticTokens;
 import org.eclipse.jdt.ls.core.internal.semantictokens.SemanticTokensVisitor;
+import org.eclipse.jdt.ls.core.internal.semantictokens.TokenModifier;
+import org.eclipse.jdt.ls.core.internal.semantictokens.TokenType;
+import org.eclipse.lsp4j.SemanticTokens;
+import org.eclipse.lsp4j.SemanticTokensLegend;
+import org.eclipse.lsp4j.SemanticTokensParams;
 
-public class SemanticTokensCommand {
-	public static SemanticTokens provide(String uri) {
-		JobHelpers.waitForJobs(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS, null);
-		return doProvide(uri);
-	}
+public class SemanticTokensHandler {
 
-	private static SemanticTokens doProvide(String uri) {
-		ITypeRoot typeRoot = JDTUtils.resolveTypeRoot(uri);
-		if (typeRoot == null) {
-			return new SemanticTokens(new int[0]);
+	public static SemanticTokens provide(IProgressMonitor monitor, SemanticTokensParams params) {
```

Comment:
`semanticTokens/full` request is requesting semantic tokens for a whole file. Suggest to rename it to something related to the request name (e.g. "semanticTokensFull" or "full"), to distinguish with `semanticTokens/range` and `semanticTokens/full/delta`.